core/src/main/java/io/grpc/internal/HttpUtil.java

```diff
@@ -44,6 +44,18 @@
 public final class HttpUtil {
 
   /**
+   * {@link io.grpc.Metadata.Key} for the :authority pseudo header.
+   */
+  public static final Metadata.Key<String> AUTHORITY_KEY =
```

Comment:
Instead of having this authority key, could we move overriding "authority" to CallOptions?

There is still a question on how to expose the authority on server-side, but that is an orthogonal issue. All users of authority today I think are using it only on client-side.